lib/elastic_apm/spies/faraday.rb

```diff
@@ -109,8 +109,14 @@ def run_request(method, url, body, headers, &block)
                   yield req if block
                 end
               rescue Faraday::ClientError, Faraday::ServerError => e # Faraday::Response::RaiseError
-                status = e.response_status if e.respond_to?(:response_status)
-                status ||= e.response&.fetch(:status)
```

Comment:
Not sure when `e.response` would be a hash, but I've preserved this behavior down below and I'm also handling a _possible_ hash response without a `:status` key as well.

The primary error I want to fix is:

```
undefined method `fetch' for #<Faraday::Response:0x000000000c657078>
```

Reply:
> Not sure when e.response would be a hash

This applies in older versions of Faraday, you can see the change [here](https://github.com/elastic/apm-agent-ruby/pull/1419) and there's more information following the corresponding issue.